### ISSUE TITLE
Flake: container 'stopped' vs 'exited'

### ISSUE THREAD

**Issue by edsantiago** [containers/libpod#5336]
I'm seeing CI flakes, a few per day, in which some test runs `ps` or `inspect`, checks `.State.Status`, expects it to be `exited`, but sees `stopped` instead.

I don't know if this is f30 or f31, runc or crun, root or rootless. I can't reproduce it at will. Rerunning the CI job usually makes the problem go away. I got nothing; I just want to file this so we can start linking to log results and gather some data.

**Comment by edsantiago** [comment]
Here's one: https://api.cirrus-ci.com/v1/task/5541879722803200/logs/integration_test.log

I can't tell which OS this is on, but it looks like APIv2 testing.

**Comment by mheon** [comment]
This is potentially a timing issue. It requires that Podman have successfully cleaned up the container before the status call comes in, which is probably happening in parallel in the background. CLI should have checks built in for this, but APIv2 might be missing them.

**Comment by edsantiago** [comment]
Here's [an example](https://cirrus-ci.com/task/5167712590626816). As @mheon predicted, it's in the APIv2 test.

**Comment by baude** [comment]
I believe you will need to test for both conditions.  As @mheon explained it to me, those cannot be predicted as easy.  In our tests, we have done somethink like isStopped which is stopped || exited.

**Comment by adelton** [comment]
For the record, I've never experienced this before -- I was always getting `exited` in all https://github.com/freeipa/freeipa-container runs.

Then about ten days ago I started to reliably get `stopped` on Travis CI Ubuntus 20.04, then two days ago it started to happen on GitHub Actions' Ubuntu 20.04. I was also able to reproduce getting the `stopped` on local Ubuntu 20.04 VM. On Fedora 33 I still get `exited`, with the same steps

```
podman run --name test -t registry.centos.org/centos:8 sleep 5
podman inspect test '--format={{.State.Status}}'
```

**Comment by mheon** [comment]
There are two potential causes. Firstly, there could be a race - Podman is taking longer than usual to clean up the container and transition to Exited. Secondly, the Podman cleanup process might not be firing (or otherwise is experiencing an error), so the container is not being unmounted and having its network setup torn down after it exits. The second is a serious bug. We did recently catch a case where that could happen due to a Conmon coredump, and made a new Conmon release to fix it.

**Comment by adelton** [comment]
@mheon Do you have a pointer to the reproducer for that conmon coredump issue? I might be able to check if that's what is happening on Ubuntu.

**Comment by mheon** [comment]
Unfortunately, I don't think we had a solid reproducer - we noticed that the RHEL 8.4 test suite was dumping core on several tests, traced it to Conmon, and fixed it from said coredumps without really looking further.The fix was in https://github.com/containers/conmon/pull/233, and the fixed release was 2.0.25.
